src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -270,27 +286,42 @@ internal string FindBestVersionMatch(VersionRange versionRange, IEnumerable<stri
                 return matchingVersion?.ToString();
             }
 
-            // Check to see if there is a max bundle version set via hosting configuration, if yes then use that instead of the one
-            // available on VM or local machine. Only use MaximumBundleV3Version or MaximumBundleV4Version if the version configured
-            // by the customer resolved to version higher than the version set via hosting config.
-            if (!string.IsNullOrEmpty(configOption.MaximumBundleV3Version)
-                && matchingVersion?.Major == ScriptConstants.ExtensionBundleV3MajorVersion)
+            return matchingVersion?.ToString();
+        }
+
+        // Limit the bundle version range so that it is higher than 4.2.0 and lower than 4.22.0
+        private VersionRange GetCappedVersion(int dotnetVersion, VersionRange versionRange, string bundleId)
+        {
+            if (dotnetVersion == ScriptConstants.DotNetVersionSix
+                    && bundleId == ScriptConstants.DefaultExtensionBundleId
+                    && versionRange.MinVersion.Major == ScriptConstants.ExtensionBundleV4MajorVersion)
             {
-                var maximumBundleV3Version = NuGetVersion.Parse(configOption.MaximumBundleV3Version);
-                matchingVersion = matchingVersion > maximumBundleV3Version ? maximumBundleV3Version : matchingVersion;
-                return matchingVersion?.ToString();
+                NuGetVersion cappedMinversion = new(ScriptConstants.CappedMinimumBundleV4VersionExclusive);
+                NuGetVersion cappedMaxVersion = new(ScriptConstants.CappedMaximumBundleV4VersionExclusive);
```

Comment:
Can we swittch the "cap/capped" terminology here (drop where not needed, switch minVersion/maxVersion below to something like `resolvedMinVersion` or `effectiveMinVersion`)